[Malcom Tan] iP

DannyDakota opened the pull request:
# Whey, track your tasks with ease!
>If you fail to plan, you are planning to fail! - Benjamin Franklin

## Do you fall under these categories?
- Can't seem to get any work [DONE](https://theconversation.com/what-happens-to-your-body-when-youre-stressed-81789)?
- Always falling behind 😞 ?
- You want to attain dean's list 👍 ?

Well fret not, introducing **Whey**

## *Why* whey? Because:
1. Text-based, easy to use
2. Fun and interactive interface
3. Just because

## Features:
- [x] Managing Tasks
- [ ] Managing ~Deadline~ Deadlines
- [ ] Reminders(coming soon! hehe)

## Here's the main method:
```java
public static void main(String[] args) {
        String pathToFile = "./data/duke.txt";
        new Duke(pathToFile).run();
    }
```

Commit:
Add Gradle support

DannyDakota committed:
Level 1 - Greet, Echo, Exit

DannyDakota committed:
Level-2 added List State and error checking for whitespace characters

DannyDakota committed:
Level-3 added Mark and Unmark feature, Task class

DannyDakota committed:
Level-4 added Deadline, ToDo and Event class, dealt with specific tasks in main function

DannyDakota committed:
Level-4 fixed system out error for ToDo

DannyDakota committed:
A-TextUiTesting Finished testing scripts to check for correctness of program

DannyDakota committed:
A-TextUiTesting (2nd commit) edited test scripts to include list

DannyDakota committed:
Level-5 Handle errors, added DukeException class to test validity of sysin

DannyDakota committed:
Level-6 Delete. Also changed bot strings

DannyDakota committed:
add save to harddisk feature

DannyDakota committed:
add Date and Time feature to event and deadline class

DannyDakota committed:
Merge branch 'branch-level-8'

DannyDakota committed:
add moreOOP classes

DannyDakota committed:
package classes into different packages A-packages

DannyDakota committed:
add test for todo, event and tasklist

DannyDakota committed:
build jar file

DannyDakota committed:
A-JavaDoc

DannyDakota committed:
A-JavaDoc

DannyDakota committed:
A-CodingStandard

DannyDakota committed:
Merge remote-tracking branch 'origin/branch-A-JavaDoc' into branch-A-JavaDoc

DannyDakota committed:
A-JavaDoc

DannyDakota committed:
Level-9

DannyDakota committed:
Merge branch 'branch-Level-9'

DannyDakota committed:
Level-9

DannyDakota committed:
Level-9

DannyDakota committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

DannyDakota committed:
instantiate gradle to build and run iP

DannyDakota committed:
Add checkstyle and amend code for checkstyle violations

pyk595 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,65 @@
+import duke.ui.Ui;
+
+/**
+ * Duke class
```
Maybe this JavaDoc comment can be more descriptive? You might want to consider putting more information on the behaviour and intended usage

pyk595 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * run the program
```
Same as above, can consider being more specific on what "run the progam" entails, despite us all knowing the general gist of what the program is supposed to do

pyk595 commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,21 @@
+public class ByeCommand extends Command<String> {
+
+    /**
+     * Constructor for ByeCommand object
```
Can consider changing to "Constructs the ByeCommand object"? To follow the active voice style showed in tutorial and the documentation

pyk595 commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,21 @@
+    }
+
+    /**
+     * change the status of the program
```
Might want to change "change" to "changes"? As per the reasoning above

pyk595 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,31 @@
+     * change status of program
+     * @param bool boolean to change the status of program
+     */
+    public void changeRunning(boolean bool) {
```
Might want to use a more descriptive variable name for bool? maybe something like "isRunning" to follow the coding standard

pyk595 commented on src/test/java/duke/task/EventTest.java:
```diff
@@ -0,0 +1,18 @@
+
+public class EventTest {
+    @Test
+    public void TimeParser_WithoutDisk_expectedBehavior() {
```
Might want to change your test method names to camel case instead of pascal case?

pyk595 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,80 @@
+    /**
+     * System out loading error
+     */
+    public void showLoadingError() {
```
I like your naming of methods in this class, very clear that they are verbs!

pyk595 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,40 @@
+public class Task {
+
+    private String description;
+    private boolean done;
```
Can consider changing your boolean variable to isDone?

pyk595 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,40 @@
+    public void setDone(boolean bool) {
+        this.done = bool;
+    }
+    @Override
```
Can consider putting JavaDoc comments for overridden methods because you are changing the expected behaviour?

pyk595 reviewed (commented):
Generally speaking most of the coding standard is adhered to, just a bit of tweaks here and there such as more descriptive javadoc comments for classes and boolean naming. Overall good job done though!

DannyDakota committed:
Level-10

DannyDakota committed:
Level-10

DannyDakota committed:
A-Assertions

DannyDakota committed:
A-Assertions

DannyDakota committed:
add assert to important points in code

DannyDakota committed:
add assert to important points in code

DannyDakota committed:
Merge remote-tracking branch 'origin/branch-A-Assertions' into branch-A-Assertions

DannyDakota committed:
Merge branch 'branch-A-Assertions'

DannyDakota committed:
Refactor code to conform to code quality standards

Code consisted of poor naming conventions and nested methods

Code should conform to coding standards

Refactoring names and abstracting out nested methods

To allow for greater readability

DannyDakota committed:
Merge pull request #1 from DannyDakota/branch-A-CodeQuality

Refactor code to conform to code quality standards

DannyDakota committed:
Add C-Priority extension

Program lacked extension

Program should have Priority extension to rank tasks according to priority

Refactoring of existing constructors to allow for priority

To allow for greater application functionality

DannyDakota committed:
A-BetterGui

DannyDakota committed:
Changing bot name

DannyDakota committed:
fix styling errors and add background

DannyDakota committed:
A-UserGuide

DannyDakota committed:
update README.md
